cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/DraftStore.java

```diff
@@ -0,0 +1,300 @@
+   * @param sortRequest {@link SortRequest} describing the sort to be performed
+   * @return a sorted list of {@link StructuredRow}
+   */
+  private List<StructuredRow> doSort(List<StructuredRow> rows, SortRequest sortRequest) {
+    if (sortRequest == null) {
+      return rows;
+    }
+    String sortField = sortRequest.getFieldName();
+    Map<String, FieldType.Type> fieldTypes = TABLE_SPEC
+      .getFieldTypes()
+      .stream()
+      .collect(Collectors.toMap(FieldType::getName, FieldType::getType));
+    if (!fieldTypes.containsKey(sortField)) {
+      throw new IllegalArgumentException(
+        String
+          .format("Invalid value '%s' for sortBy. This field does not exist in the Drafts table.",
+                  sortField));
+    }
+
+    rows.sort((o1, o2) -> {
+      int result = 0;
+      FieldType.Type fieldType = fieldTypes.get(sortField);
+
+      Object left = null;
+      Object right = null;
+      switch (fieldType) {
+        case STRING:
+          left = o1.getString(sortField);
+          right = o2.getString(sortField);
+          break;
+        case INTEGER:
+          left = o1.getInteger(sortField);
+          right = o2.getInteger(sortField);
+          break;
+        case LONG:
+          left = o1.getLong(sortField);
+          right = o2.getLong(sortField);
+          break;
+        case FLOAT:
+          left = o1.getFloat(sortField);
+          right = o2.getFloat(sortField);
+          break;
+        case DOUBLE:
+          left = o1.getDouble(sortField);
+          right = o2.getDouble(sortField);
+          break;
+        case BYTES:
+          throw new IllegalArgumentException("Cannot sort field '%s' because it is of type BYTES.");
+      }
+      if (left == null && right == null) {
+        return 0;
+      }
+      if (left == null || right == null) {
+        result = left == null ? -1 : 1;
+      } else {
+        result = ((Comparable<Object>) left).compareTo(right);
+      }
+
+      if (sortRequest.getOrder() == SortRequest.SortOrder.ASC) {
+        return result;
+      }
+      return -1 * result;
+    });
+
+    return rows;
+  }
+
+  /**
+   * Fetch a given draft if it exists
+   *
+   * @param id {@link DraftId} that is used to uniquely identify a draft
+   * @return an {@link Optional<Draft>} representing the requested draft
+   */
+  public Optional<Draft> getDraft(DraftId id) throws TransactionException {
+    AtomicReference<Optional<Draft>> result = new AtomicReference<>();
+    transactionRunner.run(context -> {
+      StructuredTable table = context.getTable(TABLE_ID);
+      Optional<StructuredRow> row = table.read(getKey(id));
+      result.set(row.map(this::fromRow));
+    });
+    return result.get();
+  }
+
+  /**
+   * Delete the given draft. This is a no-op if the draft does not exist
+   *
+   * @param id {@link DraftId} that is used to uniquely identify a draft
+   */
+  public void deleteDraft(DraftId id) throws TransactionException {
```

Comment:
throwing the TransactionException is not a very good contract. You should catch the TransactionException and extract the cause from it and throw that one out.

Reply:
Done